src/Analysis/Engine/Impl/AnalysisValue.cs

```diff
@@ -67,23 +71,25 @@ public class AnalysisValue : IAnalysisValueOperations {
 
         public virtual IEnumerable<OverloadResult> Overloads => Enumerable.Empty<OverloadResult>();
 
-        public virtual string Description {
-            get {
-                var hrd = this as IHasRichDescription;
-                if (hrd != null) {
-                    return string.Join("", hrd.GetRichDescription().Select(kv => kv.Value));
-                }
-                return null;
-            }
-        }
+        public virtual string Description
+            => this is IHasRichDescription hrd
+                ? string.Join("", hrd.GetRichDescription().Select(kv => kv.Value))
+                : string.Empty;
 
         public virtual string ShortDescription {
             get {
-                var hrd = this as IHasRichDescription;
-                if (hrd != null) {
-                    return string.Join("", hrd.GetRichDescription().TakeWhile(kv => kv.Key != WellKnownRichDescriptionKinds.EndOfDeclaration).Select(kv => kv.Value));
+                if (string.IsNullOrEmpty(_shortDescription) && this is IHasRichDescription hrd) {
+                    var sb = new StringBuilder();
+                    foreach (var item in hrd.GetRichDescription().TakeWhile(kv => kv.Key != WellKnownRichDescriptionKinds.EndOfDeclaration).Select(kv => kv.Value)) {
+                        if (sb.Length >= MaxShortDescriptionLength) {
+                            sb.Append("...");
```

Comment:
Is it better than unicode ellipsis (https://www.fileformat.info/info/unicode/char/2026/index.htm)?

Reply:
IMO probably, since it's also the same string for Python's ellipsis, and clients will typically render `...` how they see fit depending on the context (though in our case, this will come out to be monospaced).

Reply:
Existing code just does plain string - I think because doc text from Python uses plain string as well so appearance of `...` in tooltips etc will be consistent since otherwise some`...` will be different from others in UI appearance.